unittests/api_tests.cpp

```diff
@@ -437,6 +437,36 @@ BOOST_FIXTURE_TEST_CASE(action_tests, TESTER) { try {
    BOOST_REQUIRE_EQUAL( validate(), true );
 } FC_LOG_AND_RETHROW() }
 
+// test require_recipient loop (doesn't cause infinite loop)
+BOOST_FIXTURE_TEST_CASE(require_notice_tests, TESTER) { try {
+      produce_blocks(2);
+      create_account( N(testapi) );
+      create_account( N(acc5) );
+      produce_blocks(1);
+      set_code( N(testapi), test_api_wast );
+      set_code( N(acc5), test_api_wast );
+      produce_blocks(1);
+
+      // test require_notice
+      auto scope = std::vector<account_name>{N(testapi)};
+      auto test_require_notice = [this](auto& test, std::vector<account_name>& scope){
```

Comment:
Does the lambda actually get you anything here?  If not simplify it and just make it part of the method.